src/dashboard/apps/core/models.py

```diff
@@ -312,6 +312,17 @@ def get_linked_stations(self) -> dict[str, list[str]] | None:
 
         return dict(stations_grouped)
 
+    @property
+    def previous_renewable(self):
+        """Get the most recent renewable meter reading for this delivery point.
+
+        Returns:
+            QuerySet: A queryset containing the last renewable,
+                    ordered by collected_at descending. Use .first() or [0] to access
+                    the single object.
+        """
+        return self.renewables.order_by("-collected_at")
```

Comment:
You may be more opinionated:

```suggestion
        return self.renewables.order_by("-collected_at").first()
```